apps/web-mzima-client/src/app/settings/surveys/survey-item/survey-item.component.ts

```diff
@@ -344,27 +348,57 @@ export class SurveyItemComponent extends BaseComponent implements OnInit {
     }
   }
 
-  public async openConfirmModal() {
-    if (this.hasChanges()) {
+  @HostListener('window:beforeunload', ['$event']) unloadNotification($event: any): void {
+    const formHasChanged = this.hasChanges();
+    if (formHasChanged)
+      $event.returnValue = 'Form has changed! - Just to let browser popup show up';
+  }
+
+  @HostListener('window:popstate', ['$event']) onPopState() {
+    this.browserBackButtonClicked = true;
+  }
+
+  public async openConfirmModal({
```

Comment:
Move this whole functionality to the guard, you don't need it in the survey-item component

Reply:
Ok.